base/array.jl

```diff
@@ -563,6 +563,12 @@ convert(::Type{Array{T,n}}, x::AbstractArray{S,n}) where {T,n,S} = copy!(Array{T
 
 promote_rule(a::Type{Array{T,n}}, b::Type{Array{S,n}}) where {T,n,S} = el_same(promote_type(T,S), a, b)
 
+# constructors should make copies
```

Comment:
Is this documented somewhere? This is something I wondered when writing CategoricalArrays contructors. Since one-argument constructors fall back to `convert`, by default they do not make copies. I agree it would make sense for `convert`not to make a copy and for constructors to make a copy, but this should be stated clearly in the manual. Then we should review other types in Base (and in our packages) to check that this rule is followed consistently (not only for arrays).

Reply:
I'll re-do the manual entries on convert and construct as part of #15120. In progress in #23273.